src/components/safe-apps/TransactionQueueBar/index.tsx

```diff
@@ -0,0 +1,74 @@
+import { ReactElement, useState } from 'react'
+import { Backdrop, Typography, Box, IconButton, Accordion, AccordionDetails, AccordionSummary } from '@mui/material'
+import { ClickAwayListener } from '@mui/base'
+import CloseIcon from '@mui/icons-material/Close'
+import ExpandLessIcon from '@mui/icons-material/ExpandLess'
+
+import useTxQueue from '@/hooks/useTxQueue'
+import PaginatedTxns from '@/components/common/PaginatedTxns'
+import styles from './styles.module.css'
+import { getQueuedTransactionCount } from '@/utils/transactions'
+
+const TransactionQueueBar = (): ReactElement | null => {
+  const [expanded, setExpanded] = useState(false)
+  const [dismissedByUser, setDismissedByUser] = useState(false)
+
+  const toggleQueueBar = (): void => {
+    setExpanded((prev) => !prev)
+  }
+
+  const { page = { results: [] } } = useTxQueue()
+  const queuedTxCount = getQueuedTransactionCount(page)
+
+  return !dismissedByUser ? (
+    <>
+      <Box className={styles.barWrapper}>
+        <ClickAwayListener onClickAway={() => setExpanded(false)} mouseEvent="onMouseDown" touchEvent="onTouchStart">
+          <Accordion
+            data-testid="transaction-queue-bar"
+            expanded={expanded}
+            onChange={toggleQueueBar}
+            TransitionProps={{
+              timeout: {
+                appear: 400,
+                enter: 0,
+                exit: 500,
+              },
+              unmountOnExit: true,
+              mountOnEnter: true,
+            }}
+          >
+            <AccordionSummary
+              data-testid="transaction-queue-bar-summary"
+              sx={{ '.MuiAccordionSummary-content': { alignItems: 'center' } }}
+            >
+              <Typography variant="body1" color="primary.main" fontWeight={700} sx={{ mr: 'auto' }}>
+                ({queuedTxCount}) Transaction Queue
+              </Typography>
+
+              <IconButton
+                onClick={(event) => {
+                  event.stopPropagation()
+                  toggleQueueBar()
+                }}
+                aria-label="close transaction queue bar"
+                sx={{ transform: expanded ? 'rotate(180deg)' : undefined }}
+              >
+                <ExpandLessIcon />
+              </IconButton>
+              <IconButton onClick={() => setDismissedByUser(true)} aria-label="dismiss transaction queue bar">
+                <CloseIcon />
+              </IconButton>
+            </AccordionSummary>
+            <AccordionDetails>
+              <PaginatedTxns useTxns={useTxQueue} disableTopActionMargins />
```

Comment:
> disableTopActionMargins

Can we just give the list a top margin here to compensate for the negative margin inside? Passing a prop just for a minor style detail seems a bit excessive.

(I realize we probably shouldn't have used negative margins in the first place).

Reply:
Theoretically, we could, but this would require making the transaction bar component heavily dependent on the implementation details of `PaginatedTxns`, so if you change the margins in there, you also have to adjust them in the bar. So I think this is no good.

An alternative solution could be explicitly passing the styles for the top action, so each component consumer sets their own. What do you think?

Reply:
True, valid point. Let's keep the prop for now then, I'll see if we can get rid of the negative margins.